### ISSUE TITLE
[BUG] - Default groups misadvertised in CLI?

### ISSUE THREAD

**Issue by krassowski** [nebari-dev/nebari#2363]
### Describe the bug

The CLI help while technically true, does not reflect all the groups to which new user is added:

https://github.com/nebari-dev/nebari/blob/3e1dfde9c56cfe6c151b36333713d6fbea30bf86/src/_nebari/subcommands/keycloak.py#L40-L41

It seems that the user also gets added to `/developer` group:

https://github.com/nebari-dev/nebari/blob/3e1dfde9c56cfe6c151b36333713d6fbea30bf86/src/_nebari/keycloak.py#L39-L53

```
Run nebari keycloak adduser --user "${TEST_USERNAME}" "${TEST_PASSWORD}" --config nebari-config.yaml
Created user=test-user
1 Keycloak Users
username                         | email                            | groups
------------------------------------------------------------------------------------------------------------------------
test-user                        | test-user@github-actions.nebari.dev | ['analyst', 'developer', 'users']
```

### Expected behavior

The CLI help aligns with reality

### OS and architecture in which you are running Nebari

-

### How to Reproduce the problem?

See above

### Compute environment

None

**Comment by viniciusdc** [comment]
Good catch @krassowski ! I know what might be happening... Keycloak has a setting for automatically binding new users to a default group (which in this case is `Analyst`), and the keycloak command is adding the `Developer` group to them, so as an extra check, it might be worth creating an user on Keycloak and one with the command and comparing them. 

obs. I think this Keycloak default group mapping is only enabled as a custom change on `nebari.quansight`, might not be present as a default

**Comment by krassowski** [comment]
> Keycloak has a setting for automatically binding new users to a default group (which in this case is Analyst),

It looks like this is configurable:

https://github.com/nebari-dev/nebari/blob/a26b8df19c31363082ca3104e4e7b2e9354b22ae/src/_nebari/stages/kubernetes_keycloak_configuration/__init__.py#L18-L19

So there are two problems:
- CLI help is saying something which usually is true but may not be true if defaults are changed (which is not a big deal)
- CLI help does not mention the `developer` group, even though it is the CLI which adds it by default and has full control over it; this can lead to user receiving higher privileges than expected and IMO should be either clarified in the CLI help or changed
